builtin/providers/aws/resource_aws_emr_instance_group.go

```diff
@@ -0,0 +1,219 @@
+package aws
+
+import (
+	"errors"
+	"log"
+
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/emr"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+var emrInstanceGroupNotFound = errors.New("No matching EMR Instance Group")
+
+func resourceAwsEMRInstanceGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsEMRInstanceGroupCreate,
+		Read:   resourceAwsEMRInstanceGroupRead,
+		Update: resourceAwsEMRInstanceGroupUpdate,
+		Delete: resourceAwsEMRInstanceGroupDelete,
+		Schema: map[string]*schema.Schema{
+			"cluster_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"instance_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"instance_count": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  0,
+			},
+			"running_instance_count": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"status": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsEMRInstanceGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).emrconn
+
+	clusterId := d.Get("cluster_id").(string)
+	instanceType := d.Get("instance_type").(string)
+	instanceCount := d.Get("instance_count").(int)
+	groupName := d.Get("name").(string)
+
+	params := &emr.AddInstanceGroupsInput{
+		InstanceGroups: []*emr.InstanceGroupConfig{
+			{
+				InstanceRole:  aws.String("TASK"),
+				InstanceCount: aws.Int64(int64(instanceCount)),
+				InstanceType:  aws.String(instanceType),
+				Name:          aws.String(groupName),
+			},
+		},
+		JobFlowId: aws.String(clusterId),
+	}
+
+	log.Printf("[DEBUG] Creating EMR task group params: %s", params)
+	resp, err := conn.AddInstanceGroups(params)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] Created EMR task group finished: %#v", resp)
+	d.SetId(*resp.InstanceGroupIds[0])
```

Comment:
is it worth checking that InstanceGroupsIds isn't nil before dereferencing here?